tests/testImageDecorrelation.py

```diff
@@ -281,23 +310,66 @@ def _testImages(self):
+        corrected_diffExp = self._runDecorrelationTask(diffExp, mKernel)
+        self._testDecorrelation(expected_var, corrected_diffExp)
 
-    def testDiffimCorrection_higher_science_variance(self):
-        """Test decorrelated diffim when the science image variance is higher than that of the template.
+    def testDiffimCorrection_same_variance(self):
+        """Test decorrelated diffim from images with different variances.
         """
-        self._setUpImages(svar=0.08, tvar=0.04)
-        self._testImages()
-
-    def testDiffimCorrection_higher_template_variance(self):
-        """Test decorrelated diffim when the template variance is higher than that of the science img.
+        # Same variance
+        self._testDiffimCorrection(svar=0.04, tvar=0.04)
+        # Science image variance is higher than that of the template.
+        self._testDiffimCorrection(svar=0.08, tvar=0.04)
+        # Template variance is higher than that of the science img.
+        self._testDiffimCorrection(svar=0.04, tvar=0.08)
+
+    def _runDecorrelationTaskMapReduced(self, diffExp, mKernel):
+        """ Run decorrelation using the imageMapReducer.
+        """
+        config = DecorrelateALKernelMapperConfig()
+        config.borderSizeX = config.borderSizeY = 3
+        config.reducerSubtask.reduceOperation = 'average'
+        task = ImageMapReduceTask(config=config)
+        decorrResult = task.run(diffExp, template=self.im2ex, science=self.im1ex,
+                                psfMatchingKernel=mKernel, forceEvenSized=True)
+        corrected_diffExp = decorrResult.exposure
+        return corrected_diffExp
+
+    def _testDiffimCorrection_mapReduced(self, svar, tvar, varyPsf=0.0):
+        """ Run decorrelation using the imageMapReduce task, and check the variance of
+            the corrected diffim.
+        """
+        self._setUpImages(svar=svar, tvar=tvar, varyPsf=varyPsf)
+        diffExp, mKernel, expected_var = self._makeAndTestUncorrectedDiffim()
+        corrected_diffExp = self._runDecorrelationTaskMapReduced(diffExp, mKernel)
+        self._testDecorrelation(expected_var, corrected_diffExp)
+        # Also compare the diffim generated here vs. the non-ImageMapReduce one
+        corrected_diffExp_OLD = self._runDecorrelationTask(diffExp, mKernel)
+        self.assertImagesNearlyEqual(corrected_diffExp.getMaskedImage().getImage(),
+                                     corrected_diffExp_OLD.getMaskedImage().getImage())
+        self.assertImagesNearlyEqual(corrected_diffExp.getMaskedImage().getVariance(),
+                                     corrected_diffExp_OLD.getMaskedImage().getVariance())
+        self.assertMasksEqual(corrected_diffExp.getMaskedImage().getMask(),
+                              corrected_diffExp_OLD.getMaskedImage().getMask())
+        # Is this the same as the above? Let's just be complete.
+        self.assertMaskedImagesNearlyEqual(corrected_diffExp.getMaskedImage(),
```

Comment:
I think you only need the last line of these: see the docstring, which says "The mask planes must match exactly. The image and variance planes are nearly equal if all pixels obey: |val1 - val0| <= rtol*|val1| + atol or, for float types, if nan/inf/-inf pixels match.":

https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1afw_1_1image_1_1test_utils.html#abf24e24b9b36b6432943b513f2739822